subsys/pm/device.c

```diff
@@ -44,6 +59,16 @@ int pm_device_state_set(const struct device *dev,
 		}
 
 		action = PM_DEVICE_ACTION_SUSPEND;
+
+		/*
+		 * Power domain will be OFF, now notify all dependents
+		 * about it
+		 */
+		if (domain) {
+			(void)device_supported_foreach(dev,
+			      domain_supported_cb,
+			      (void *)PM_DEVICE_ACTION_DOMAIN_OFF);
```

Comment:
nit: why is error code ignored here?

Reply:
The problem is what do with another error.

If the domain is resuming the action happens before notify devices, in this case, if a device returns an error we would need to suspend the domain again ? And notify all devices that were already notified ?

If the domain is supending, what to do, cancel the operation and notify devices that received the notification that the domain will continue active ?

Also, this will require all devices properly return -ENOTSUP or 0 when not handling these notifications. I don't know if initially we should be doing it, but I can change it though.